pkg/driverbuilder/kubernetes.go

```diff
@@ -270,16 +257,31 @@ func (bp *KubernetesBuildProcessor) copyModuleAndProbeFromPodWithUID(ctx context
 			if p.Status.Phase == corev1.PodRunning {
 				logger.WithField(falcoBuilderUIDLabel, falcoBuilderUID).Info("start downloading module and probe from pod")
 				if builder.ModuleFullPath != "" {
+					outModule, err := os.Create(build.ModuleFilePath)
+					if err != nil {
+						return err
+					}
+
 					err = copySingleFileFromPod(outModule, bp.coreV1Client, bp.clientConfig, p.Namespace, p.Name, builder.ModuleFullPath)
 					if err != nil {
 						return err
 					}
+					outModule.Close()
+					logger.Info("Kernel Module extraction successful")
 				}
 				if builder.ProbeFullPath != "" {
+					time.Sleep(15 * time.Second)
+					outProbe, err := os.Create(build.ProbeFilePath)
+					if err != nil {
+						return err
+					}
```

Comment:
I'd defer the Close() here; you are leaking it if copySingleFileFromPod fails below.